SaneEconomyCore/src/main/java/org/appledash/saneeconomy/BukkitBootstrap.java

```diff
@@ -0,0 +1,132 @@
+            throw new RuntimeException("Could not load required deps.");
+        }
+
+        concrete = new SaneEconomy(this);
+
+        concrete.onLoad();
+    }
+
+    @Override
+    public void onEnable() {
+        concrete.onEnable();
+    }
+
+    @Override
+    public void onDisable() {
+        concrete.onDisable();
+    }
+
+    private void loadJars(File jarsFolder, URLClassLoader classLoader) throws IOException, IllegalAccessException, InvocationTargetException {
+        if (jarsFolder.exists() && !jarsFolder.isDirectory()) {
+            Files.delete(jarsFolder.toPath());
+        }
+        if (!jarsFolder.exists()) {
+            if (!jarsFolder.mkdirs()) {
+                throw new IOException("Could not create parent directory structure.");
+            }
+        }
+
+        log(Level.INFO, LogUtil.getHeading() + ChatColor.YELLOW + "Loading dep " + ChatColor.WHITE + "HikariCP");
+        JarUtil.loadJar("http://central.maven.org/maven2/com/zaxxer/HikariCP/3.2.0/HikariCP-3.2.0.jar",
+                new File(jarsFolder, "HikariCP-3.2.0.jar"),
+                new File(jarsFolder, "HikariCP-3.2.0-relocated.jar"),
+                classLoader,
+                Collections.singletonList(new Relocation(parse("com{}zaxxer{}hikari"), parse(shadedPath + "{}hikari"))));
+
+        try {
+            Class.forName("org.sqlite.JDBC", false, classLoader);
+        } catch (ClassNotFoundException ignored) {
+            log(Level.INFO, LogUtil.getHeading() + ChatColor.YELLOW + "Loading dep " + ChatColor.WHITE + "SQLite");
+            JarUtil.loadJar("http://central.maven.org/maven2/org/xerial/sqlite-jdbc/3.25.2/sqlite-jdbc-3.25.2.jar",
+                    new File(jarsFolder, "sqlite-jdbc-3.25.2.jar"),
+                    classLoader);
+
+            try {
+                DriverManager.registerDriver((Driver) Class.forName("org.sqlite.JDBC", true, classLoader).newInstance());
+            } catch (ClassNotFoundException | InstantiationException | SQLException ex) {
+                logger.error(ex.getMessage(), ex);
+            }
+        }
+
+        try {
+            Class.forName("com.mysql.jdbc.Driver", false, classLoader);
+        } catch (ClassNotFoundException ignored) {
+            log(Level.INFO, LogUtil.getHeading() + ChatColor.YELLOW + "Loading dep " + ChatColor.WHITE + "MySQL");
+            JarUtil.loadJar("http://central.maven.org/maven2/mysql/mysql-connector-java/8.0.13/mysql-connector-java-8.0.13.jar",
+                    new File(jarsFolder, "mysql-connector-java-8.0.13.jar"),
+                    classLoader);
+
+            try {
+                DriverManager.registerDriver((Driver) Class.forName("com.mysql.jdbc.Driver", true, classLoader).newInstance());
+            } catch (ClassNotFoundException | InstantiationException | SQLException ex) {
+                logger.error(ex.getMessage(), ex);
+            }
+        }
+    }
+
+    // Because Maven's relocate is maybe sometimes a bit too powerful ;)
```

Comment:
Explain further!

Reply:
Basically using Maven't relocate feature will also modify strings in the source. For example, if you relocate "com.google" to "extern.com.google" it'll find any string "com.google" in the code and also transform that. Obviously this will completely wreck the jar-relocator code, so to get around this you replace the "." with a "{}" in the string literal and then replace with a period again at runtime.

Reply:
(so maybe put that in the comment)